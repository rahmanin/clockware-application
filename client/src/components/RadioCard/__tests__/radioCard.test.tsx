import {RadioCard} from "../index"
import React from 'react';
import {cleanup} from '@testing-library/react';
import { act } from "react-dom/test-utils";
import {shallow} from "enzyme"
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

Enzyme.configure({ adapter: new Adapter() });
afterEach(cleanup);

act(() => {
  it('Render RadioCard', () => {
    const component = shallow(
        <RadioCard 
          onChange={()=>{return}} 
          name={"string"} 
          precision={1} 
          onClickFeedbacks={()=>{return}}
          onClickShowAll={()=>{return}} 
          onVisibleChange={()=>{return}}
        />
    );
    expect(component).toMatchSnapshot();
  });
})
