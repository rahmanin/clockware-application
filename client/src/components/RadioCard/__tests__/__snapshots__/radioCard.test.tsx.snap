// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render RadioCard 1`] = `
<label
  className="radio_label"
>
  <input
    className="card-input-element"
    name="string"
    onChange={[Function]}
    type="radio"
  />
  <div
    className="panel panel-default card-input"
  >
    <div
      className="panel-heading"
    />
    <div
      className="panel-body"
    >
      <RatingStars
        precision={1}
        readOnly={true}
      />
    </div>
    <div
      onClick={[Function]}
    >
      <Popover
        content={
          <React.Fragment>
            No feedbacks
            <div
              className="show_all_button"
              onClick={[Function]}
            >
              Show all
            </div>
          </React.Fragment>
        }
        mouseEnterDelay={0.1}
        mouseLeaveDelay={0.1}
        onVisibleChange={[Function]}
        overlayStyle={Object {}}
        placement="bottom"
        title="Feedbacks"
        transitionName="zoom-big"
        trigger="click"
      >
        <Button
          block={false}
          className="feedback_button"
          ghost={false}
          htmlType="button"
          loading={false}
        >
          Show feedbacks
        </Button>
      </Popover>
    </div>
  </div>
</label>
`;
